### PR TITLE
fix closing of streams in example

### LINKED ISSUE #220: cmd examples do not work
I was trying to run cmd/server/main.go and cmd/client/main.go, however they do not work (there is a runtime error):

```
$ go run server/main.go 8383
2021/08/02 16:02:01 failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 2048 kiB, got: 416 kiB). See https://github.com/lucas-clemente/quic-go/wiki/UDP-Receive-Buffer-Size for details.
Listening. Now run: go run cmd/client/main.go /ip4/0.0.0.0/udp/8383/quic QmZ8r2Zp3A1VTitYUdQaDnw7p6251XXKHusiXp8Vj5Rwgb
2021/08/02 16:02:11 Accepted new connection from QmRmQ7FvucDy5p99HQvyyNqUeZUbdmZbVLos8YLNAvDEbH (/ip4/127.0.0.1/udp/39635/quic)
2021/08/02 16:02:11 handling conn failed: Application error 0x0
```
```
$ go run client/main.go /ip4/0.0.0.0/udp/8383/quic QmZ8r2Zp3A1VTitYUdQaDnw7p6251XXKHusiXp8Vj5Rwgb
2021/08/02 16:02:11 Dialing /ip4/0.0.0.0/udp/8383/quic
2021/08/02 16:02:11 failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 2048 kiB, got: 416 kiB). See https://github.com/lucas-clemente/quic-go/wiki/UDP-Receive-Buffer-Size for details.
2021/08/02 16:02:11 Sending: Hello world!
2021/08/02 16:02:11 Read on stream 0 canceled with error code 0
exit status 1
```

### PR DESCRIPTION
**Pull request by marten-seemann**
Fixes #220.

### CODE REVIEW

File: cmd/client/main.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ func run(raddr string, p string) error {
 	if _, err := str.Write([]byte(msg)); err != nil {
 		return err
 	}
-	if err := str.Close(); err != nil {
+	if err := str.CloseWrite(); err != nil {
```
Comment (Stebalien) [inline cmd/client/main.go:61]: We still need to call close eventually (e.g., in a defer).

File: cmd/server/main.go [hunk 2]
```diff
@@ -75,5 +75,5 @@ func handleConn(conn tpt.CapableConn) error {
 	if _, err := str.Write([]byte(data)); err != nil {
 		return err
 	}
-	return str.Close()
+	return str.CloseWrite()
```
Comment (Stebalien) [inline cmd/server/main.go:78]: ditto (that or reset)

File: cmd/client/main.go (hunk 1 above)
Comment (marten-seemann) [reply cmd/client/main.go:61]: We could, but streams are automatically closed when the connection is closed, so this is not necessary.

Comment (marten-seemann) [reply cmd/client/main.go:61]: On the other hand, this is an example, and we should probably do things the clean way. Will add the defers.

File: cmd/server/main.go (hunk 2 above)
Comment (Stebalien) [reply cmd/server/main.go:79]: Hm. Actually, in this case, shouldn't it just be a close? I mean, we've already read everything we want.

**Review by Stebalien — APPROVED** [review]

File: cmd/server/main.go (hunk 2 above)
Comment (marten-seemann) [reply cmd/server/main.go:79]: That works as well. And it's easier.

**Pull request merged**